test/docker/docker_utilities.go

```diff
@@ -29,7 +30,17 @@ func AwaitDatabaseUp(t *testing.T, composeFile string, adminContainerName string
 }
 
 func GetDatabaseListings(t *testing.T, composeFile string, influxContainerName string) []string {
-	raw := executeCommandInContainer(t, composeFile, influxContainerName, "influx", "-execute", "show databases")
-	databaseListing := strings.Split(raw, "----")[1]
-	return strings.Split(databaseListing, "\n")
-}
\ No newline at end of file
+	raw := executeCommandInContainer(t, composeFile, influxContainerName, "influx", "bucket", "list", "--json")
+	var bucketList []map[string]interface{}
+	var listing []string
+	jsonData := raw
+	err := json.Unmarshal([]byte(jsonData), &bucketList)
+	if err != nil {
+		fmt.Println("Error while decoding data ", err.Error())
```

Comment:
The pattern in Golang for doing error handling is to have `error` as a component of the return value, which can be a tuple. I would change the signature of this function to return `(error, []string)` and `return err, nil` in this case.

You would have to update callers to account for the new return value. Eventually you'll find some upstream caller that can just fail the test or panic.